RELEASE/scripts/autoscend/paths/dark_gyffte.ash

```diff
@@ -363,16 +363,31 @@ boolean bat_shouldPickSkills(int hpLeft)
 	return false;
 }
 
+boolean bat_haveEnsorcelee() // checks if you have a current Ensorceled Monster
+{
+	if(!auto_have_skill($skill[Ensorcel]))
+		return false;
+
+	return get_property("ensorcelee") != "";
+}
+
+phylum bat_ensorceledMonster() //returns phylum of current Ensorceled Monster (if you have one)
+{
+	return monster_phylum(to_monster(get_property("ensorcelee")));
+}	
+
 boolean bat_shouldEnsorcel(monster m)
 {
 	if(my_class() != $class[Vampyre] || !auto_have_skill($skill[Ensorcel]))
 		return false;
 
 	// until we have a way to tell what we already have as an ensorcelee, just ensorcel goblins
 	// to help avoid getting beaten up...
-	if(m.monster_phylum() == $phylum[goblin] && !isFreeMonster(m))
+	if(m.monster_phylum() == $phylum[goblin] && !isFreeMonster(m) && !bat_haveEnsorcelee()) //stop wasting additional Ensorcel casts once we already have an Ensorcelee
 		return true;
 
+	//code to be added for getting other types of monster (beasts / bugs presumably) where appropriate.
```

Comment:
```suggestion
	//TODO code for getting other types of monster (beasts / bugs presumably) where appropriate.
```
that way if we search for stuff that is still todo we can easily search for it by just grepping the word todo